libopenage/main/tests/gui.cpp

```diff
@@ -1,22 +1,23 @@
-// Copyright 2019-2021 the openage authors. See copying.md for legal info.
+// Copyright 2019-2023 the openage authors. See copying.md for legal info.
 
 #include "gui.h"
 
 #include <algorithm>
-#include <vector>
 #include <eigen3/Eigen/Dense>
+#include <vector>
 
 #include "gamestate.h"
-#include "../../log/log.h"
-#include "../../renderer/opengl/shader.h"
-#include "../../renderer/resources/shader_source.h"
-#include "../../renderer/resources/texture_data.h"
-#include "../../renderer/resources/mesh_data.h"
-#include "../../renderer/texture.h"
-#include "../../renderer/shader_program.h"
-#include "../../renderer/util.h"
-#include "../../renderer/geometry.h"
-#include "../../renderer/opengl/window.h"
+#include "log/log.h"
+#include "renderer/geometry.h"
+#include "renderer/opengl/context.h"
+#include "renderer/opengl/shader.h"
+#include "renderer/opengl/window.h"
+#include "renderer/resources/mesh_data.h"
+#include "renderer/resources/shader_source.h"
+#include "renderer/resources/texture_data.h"
+#include "renderer/shader_program.h"
+#include "renderer/texture.h"
+#include "renderer/util.h"
```

Comment:
why does this work? the includepath should be relative to the file, since injecting search directories requires tools to be more aware of the build system than just blindly following file paths.

Reply:
afaik you just have to forward the base include path to the compiler. I actually prefer it being an absolute path because the relative paths tend to be unreadable in error messages.

Reply:
ok well, the tooling will have to bear with it, but then we have to make sure to either use unique relative imports, and only ever to absolute imports for paths starting at `libopenage/*`